pkg/pgmodel/ingestor/inserter.go

```diff
@@ -125,9 +166,8 @@ func (p *pgxInserter) InsertData(rows map[string][]model.SamplesInfo) (uint64, e
 		for _, si := range data {
 			numRows += uint64(len(si.Samples))
 		}
-		// insertMetricData() is expected to be non-blocking,
-		// just a channel insert
-		p.insertMetricData(metricName, data, workFinished, errChan)
+		// the following is usually non-blocking, just a channel insert
+		p.getMetricInserter(metricName) <- insertDataRequest{metric: metricName, data: data, finished: workFinished, errChan: errChan}
```

Comment:
Should we be inserting a pointer? given that channels have limitations in speed for large data, which is why we pass address in prom-migrator as well.

Reply:
that's changed in PR 502